### ISSUE TITLE
[feature] Enhance Next.js README

### ISSUE THREAD

**Issue by Kinbaum** [adobe/aem-spa-project-core#38]
When following along with the readme for the Next.js remote app, the instructions are missing the following:
-  where to call `ModelManager.initializeAsync()`
- how to set up appropriate rewrites for `/.model.json` and `/content/:path*`
- how to implement the `AEMResponsiveGrid` component

When comparing this to the `aem-guides-wknd-graphql` tutorial, I am able to get it close to working using Next.js, but within the AEM environment itself, I'm running into problems where:
- I cannot drag and drop components in my responsive grid (I see nodes added to the jcr, but nothing renders on the screen).
- running into issue with rewrites (possible infinite loop). I notice this model called over and over (http://localhost:4502/content/wknd-app/us/en/home/jcr:content/root/responsivegrid.model.json)
- not properly initializing the model manager (I tried calling this in `_app.tsx` but it doesn't seem to work)

If we could get a working example of a Next.js implementation that would be great.

Here is my `pages/_app.tsx` file (assuming this is the right place to call the ModelManager)

```tsx
import { ModelClient, ModelManager } from '@adobe/aem-spa-page-model-manager';
import type { AppProps } from 'next/app';

const modelClient = new ModelClient(process.env.NEXT_PUBLIC_AEM);

ModelManager.initializeAsync({
    modelClient
});

export default function MyApp({ Component, pageProps }: AppProps) {
    return (
        <Component { ...pageProps} />
    );
}
```

In my `pages/index.tsx` I am adding the `AEMResponsiveGrid` component like so:

```tsx
export default function Home() {
    return (
        <AEMResponsiveGrid
            pagePath="/content/wknd-app/us/en/home"
            itemPath="root/responsivegrid" />
    )
}
```

> NOTE: This produces the infinite loop I mentioned above.

Here are some screenshots of what I see in the console:

<img width="637" alt="124648596-7622fe00-de65-11eb-8576-e4ee3451c4f2" src="https://user-images.githubusercontent.com/6590356/124789465-50a2fc80-df18-11eb-8013-e7629e9852c6.png">
<img width="1175" alt="124648726-99e64400-de65-11eb-8428-4842d9c8bca1" src="https://user-images.githubusercontent.com/6590356/124789493-56004700-df18-11eb-99ae-b70e5401237b.png">

I am handling rewrites in my `next.config.ts` like so:

```js
async rewrites() {
    return [
        {
            source: '/content/:path*',
            destination: '/api/proxy/:path*'
        },
        {
            source: '/.model.json',
            destination: '/api/proxy/.model.json'
        }
    ]
}
```

Produces the following error within the AEM environment
```
Uncaught (in promise) TypeError: Cannot read property 'includes' of undefined
    at resolveRewrites (resolve-rewrites.js?3fb8:7)
    at Router._callee$ (router.js?7a5c:132)
    at tryCatch (runtime.js?96cf:63)
    at Generator.invoke [as _invoke] (runtime.js?96cf:293)
    at Generator.eval [as next] (runtime.js?96cf:118)
    at asyncGeneratorStep (asyncToGenerator.js?a954:3)
    at _next (asyncToGenerator.js?a954:25)
```
I was able to get the images to pull from the DAM and use the `next/image` component with the rewrites.

**Comment by Kinbaum** [comment]
@devongovett Would you be able to assist me with this?

**Comment by Kinbaum** [comment]
@sharanyavinod Would you be able to assist me with this?

**Comment by sharanyavinod** [comment]
@Kinbaum Could you share your code package as well if possible, please?

**Comment by Kinbaum** [comment]
I am just reusing the WKND app that has the `remotenextpage`. I created a new page under `/content/wknd-app/us/en` using that resource type. Then in my next app, I’ve satisfied all of the requirements like creating the `/api/getNextProps` endpoint and `assets-manifest`. 

I’ve essentially shared all of my files from my next app. There is only a single page with a responsive grid component.  I’ll put together a zip so you can see.

**Comment by Kinbaum** [comment]
@sharanyavinod Here is a stripped down version that shows the issue when connecting to AEM.
[Code Sample.zip](https://github.com/adobe/aem-spa-project-core/files/6783296/Archive.zip)

Run `yarn` to install and `yarn dev` to run the app. Should load on localhost:3000
Once that is done, on the AEM side under the wknd-app folder (using the graphql tutorial) create a page, select the Next.js template and create the page:

<img width="1069" alt="Screen Shot 2021-07-08 at 5 30 00 AM" src="https://user-images.githubusercontent.com/6590356/124898845-8e059980-dfad-11eb-821c-3c949a3a75c7.png">

<img width="606" alt="Screen Shot 2021-07-08 at 5 32 14 AM" src="https://user-images.githubusercontent.com/6590356/124899180-dde46080-dfad-11eb-9fb7-ddc5411af1eb.png">

<img width="572" alt="Screen Shot 2021-07-08 at 5 32 47 AM" src="https://user-images.githubusercontent.com/6590356/124899264-f18fc700-dfad-11eb-823a-741bb55dad5a.png">

After running through all of those steps, open up your developer tools and the open the page just created. You should immediately begin seeing errors like the following:

<img width="1598" alt="Screen Shot 2021-07-08 at 5 35 30 AM" src="https://user-images.githubusercontent.com/6590356/124899638-51866d80-dfae-11eb-808a-52bd148f2d6e.png">

The one that specifically stands out to me is this one:
`Uncaught Error: Edit context could not be fetch due to an error in the edit context servlet`

**Comment by Kinbaum** [comment]
@sharanyavinod have you had the chance to look; any thoughts?

**Comment by sharanyavinod** [comment]
@Kinbaum Regarding the query- 
>  I cannot drag and drop components in my responsive grid (I see nodes added to the jcr, but nothing renders on the screen).
 
For container components, the child components need to be mapped in order for them to be rendered. I see that you have `MapTo` within the components itself - for eg: in the AEMText component. However, since this component isn't used, the mapping never gets executed. You could either have a common file which imports all components and performs the mappings and import them within the parent component OR just import/map possible child components within the container. 
Some References: 
https://github.com/adobe/aem-guides-wknd-graphql/blob/feature/spa-editor/react-app/src/App.js#L15
https://github.com/adobe/aem-guides-wknd-spa/blob/React/latest/ui.frontend/src/index.js#L30

Regarding the remaining errors, will take a closer look and get back to you. Thank you for the detailed descriptions and snippets!

**Comment by Kinbaum** [comment]
Thanks @sharanyavinod importing the components in my _app.tax file allowed them to show up in/outside of AEM. However in AEM, I still cannot edit due to that error. 

Also another thing I noticed is that SSR doesn’t seem to be working using these “MapTo” components. Any thoughts on that?

**Comment by sharanyavinod** [comment]
@Kinbaum Thank you for bringing this issue to our attention. Please use the newly released v1.3.8 of `spa.project.core` in your AEM project to resolve the issue.
For fetching the model.json you can use pass the AEM host to ModelClient during initialization as you've illustrated in the snippets above. The location of initialization is also as already done. I'll add this to the docs for future clarity.

Regarding SSR issues related to MapTo, if unrelated to the current issue, could you please create a separate issue with the corresponding details?

**Comment by Kinbaum** [comment]
Thank you @sharanyavinod. I have created a new issue #40 to track the SSR problem.
